sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3613,110 +3656,20 @@ class Analyzer(override val catalogManager: CatalogManager)
     private def resolveFieldNames(
         table: ResolvedTable,
         fieldName: Seq[String],
-        context: Expression): ResolvedFieldName = {
+        context: Origin): ResolvedFieldName = {
```

Comment:
the parameter type can still be `Expression`

Reply:
Reverted.